*** frxx1.f.orig	2012-05-22 15:15:17.000000000 -0700
--- frxx1.f	2012-09-07 10:37:03.000000000 -0700
***************
*** 1035,1043 ****
         KK = PTYPE(3,JF)
         IF(KK.eq.7) THEN
  C                       Take K=7 as an inelastic monopole
!           IF(IA.eq.IB) GO TO 300
            KK = 0
!           ENDIF
  C
        IF(JFT>=12.and.JFT<=13) THEN
  C                           LOOK UP TABLE OF ALLOWED COUPLINGS :
--- 1035,1046 ----
         KK = PTYPE(3,JF)
         IF(KK.eq.7) THEN
  C                       Take K=7 as an inelastic monopole
!           IF(IA.eq.IB) GO TO 300 ! exclude diagonal monopole
            KK = 0
!        ELSE if(KK==0) then
! C			Normal KK=0 should exclude off-diagonal monopoles
! 	  if(IA/=IB) GO TO 300
!        ENDIF
  C
        IF(JFT>=12.and.JFT<=13) THEN
  C                           LOOK UP TABLE OF ALLOWED COUPLINGS :
